apps/ios/Shared/Views/UserSettings/PrivacySettings.swift

```diff
@@ -96,45 +199,126 @@ struct SimplexLockSetting: View {
             case .laFailedAlert: return laFailedAlert()
             case .laUnavailableInstructionAlert: return laUnavailableInstructionAlert()
             case .laUnavailableTurningOffAlert: return laUnavailableTurningOffAlert()
+            case .laPasswordSetAlert: return passwordAlert("Passcode set!")
+            case .laPasswordChangedAlert: return passwordAlert("Passcode changed!")
+            case .laPasswordNotChangedAlert: return mkAlert(title: "Passcode not changed!")
             }
         }
+        .sheet(item: $showPasswordAction) { a in
+            switch a {
+            case .enableAuth:
+                SetAppPasscodeView {
+                    laLockDelay = 30
+                    prefPerformLA = true
+                    showChangePassword = true
+                    showLAAlert(.laPasswordSetAlert)
+                } cancel: {
+                    prefPerformLA = false
+                    withAnimation { performLA = false }
+                    performLAToggleReset = true
+                }
+            case .toggleMode:
+                SetAppPasscodeView {
+                    laLockDelay = 30
+                    currentLAMode = laMode
+                    privacyLocalAuthModeDefault.set(laMode)
+                    showChangePassword = true
+                    showLAAlert(.laPasswordSetAlert)
+                } cancel: {
+                    withAnimation { laMode = currentLAMode }
+                    performLAModeReset = true
```

Comment:
maybe the order should be inverse?
```suggestion
                    performLAModeReset = true
                    withAnimation { laMode = currentLAMode }
```

Reply:
that was like that before, I think it's processed later anyway.

Reply:
changed